cpp/create_module/algorithm/create.cpp

```diff
@@ -1 +1,26 @@
 #include "create.hpp"
+
+
+void Create::SetProperty(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    mgp::Graph graph = mgp::Graph(memgraph_graph);
+    const mgp::Node node_input = arguments[0].ValueNode();
+    const std::string string(arguments[1].ValueString());
+    mgp::Value value = arguments[2];
+    const int64_t id = node_input.Id().AsInt();
+    for (auto node : graph.Nodes()) {
+      if (node.Id().AsInt() == id) {
```

Comment:
Can you check this function:
```
inline Node Graph::GetNodeById(const Id node_id) const {
  auto mgp_node = mgp::graph_get_vertex_by_id(graph_, mgp_vertex_id{.as_int = node_id.AsInt()}, memory);
  if (mgp_node == nullptr) {
    mgp::vertex_destroy(mgp_node);
    throw NotFoundException("Node with ID " + std::to_string(node_id.AsUint()) + " not found!");
  }
  auto node = Node(mgp_node);
  mgp::vertex_destroy(mgp_node);
  return node;
}
```

I am not sure why you opted for O(n) when we can have this O(log n) complexity :smile:

Reply:
Forgot about existence of this function, will fix